### DISCUSSION TITLE
Why doesn't the compiler return non-zero code on compile error

### DISCUSSION THREAD

**Discussion by Shadlock0133** [koka-lang/koka#304] [General]
Version: 2.4.0
OS: Debian 10

E.g. as a one common way to quickly prototype code, I would use something like `compiler -o main main.lang && ./main`, which would stop before `&&` if compiler returns non-zero error code. As `koka` doesn't do it, it will run the old program regardless. Is this intentional, or a bug?

**Comment by JanBeh** [comment]
Using Koka 2.4.2, I observe the same issue. I don't know if this is intentional, but I filed a bug report:

* #350
